### PR TITLE
Feature/record builders

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
Added auto generated "with" methods to JPlagOptions, to reduce the amount of boilerplate code.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -37,6 +37,12 @@
             <version>${revision}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (dfuchss) [inline core/pom.xml:40]: Could you check if https://github.com/projectlombok/lombok could be used :)

File: core/src/test/java/de/jplag/options/JPlagOptionsTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.jupiter.api.Test;
+
+import de.jplag.java.JavaLanguage;
+
+class JPlagOptionsTest {
+    @Test
+    void testWithLanguageOption() {
+        JavaLanguage lang = new JavaLanguage();
+        JPlagOptions options = new JPlagOptions(null, Set.of(), Set.of());
+        options = options.withLanguageOption(lang);
```
Comment (dfuchss) [inline core/src/test/java/de/jplag/options/JPlagOptionsTest.java:16]: Why test a deprecated method?

File: core/pom.xml [hunk 3]
```diff
@@ -46,5 +52,15 @@
                 <directory>src/main/resources</directory>
             </resource>
         </resources>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.7.0</version>
+                <configuration>
+                    <sourcepath>src/main/java;target/generated-sources/annotations</sourcepath>
```
Comment (dfuchss) [inline core/pom.xml:61]: Please check that the source plugin also contains the generated source code of the class :)

**Review by dfuchss — APPROVED** [review]
Please check the last comment; if positive , I've approved :)

**Review by tsaglam — APPROVED** [review]

**Pull request merged**